### ISSUE TITLE
Professions imbalance and food

### ISSUE THREAD

**Issue by PacifyerGrey** [StrangeLoopGames/EcoSuggestions#473]
Having played as a lot different professions I notices some major problems.
I do understand that this mostly applies to multiplayer servers with mid to high number of players but I am sure this can be balanced to fix both single player and multiplayer.

Currently we can split professions into 2 major classes - worker professions and crafter professions.

Worker professions require player time and calories to get end result. While skills decrease both, these professions still have several pecularities:
- Very limited outcome of their end result into the economy.
- A player can not provide anything to the economy while he is offline or afk.
- Consume calories like mad.
Examples of such professions: Logging, Mining, Road building, Hunting

Crafting professions however:
- Have almost unlimited outcome. You can build a ton of crafting tables and queue many things simultaneously.
- Crafting is not interrupted when a player is offline or afk
- Do not require calories to function

This leads so several issues:
- Crafters (which a a majority) are minimally engaged in the food chain especially mid to end game
- There is not much incentive to use high tier food in the endgame
- Crafters get problems with competition as a single crafter of a given profession can provide the whole server with his services especially with the ability to queue big jobs at distant tables.

I want to propose introducing food requirements for machines.
1. Machines can require a certain food quality of the crafter using them. This is currently reflected by SP/day but you can also split it into tiers and add the number to a diet display. This will drive people to improve their diet to higher tiers driving the whole economy.
2. Machines can consume calories per minute. This will force crafters use food at all as well as effectively prevent them from super high throughput and almost disable offline activity. This will also make crafting speed skills much more valuable so they can also get a progressive SP costs same as efficiency currently have.
3. A general idea to engage a crafter in the crafting process so it would require his own time to progress. Such as some kind of a crafting mini-game at the crafting table. However this might be just too much for a single player but still. This minigame can consume those calories per action making the process look natural (crafter is actually crafting something at the table) as well as make crafting rate limited. A good example might be crafting system in Final Fantasy XIV. Surely that one is too complex for Eco but the idea makes every person be actually engaged in their activity.

All these ideas have a questionable use in single player but in multiplayer we do get following benefits:
- Require farming /cooking progression
- Require much more food so it is actually needed up until endgame where currently farmers/cooks have almost nothing to do
- Make all professions equally engaged in the process, fair competition

**Comment by mirasrael** [comment]
It is all good in general and I agree that crafting professions in their current state is unfair comparing to working professions, but it will have significant impact on gameplay. Imagine you want to build a house with eight rooms, then you need something I don't know like 500 bricks. Can you imagine somebody playing mini-game to produce these 500 bricks? And this is only for single house.

What is good and bad with worker professions - almost anyone can do them. And skills for that professions almost useless for now. So everyone has crafting profession and you can't opposite them with working professions.

The problem with high-level food is that it isn't really needed, because you can have enough skill points to cover all your needs from different sources - like housing. And another problem you mentioned - is about spending calories on high level. But while I'm liking calories spending in overall for endgame I haven't clear vision about how it will work with machines. If you have to spend calories for machine to work - should you just sit around and wait until you go out of calories, then feed yourself and wait, and repeat and repeat and repeat... Didn't you find this boring?

OK. While writing this I have an another idea. What if we will need personal for machines to work? In real world you can't just queue up production for wood working table, you have to get tools and work on it, but making this in Eco as I said before may become very boring and annoying. Instead we can imagine what your table needs personal to work (which is like invisible elves) and you should feed that personal. Then like for some tables you need coal to smelt you will need food. You will not spent calories directly, but instead you will put food into production in other way. To justify high-level food usage it may provide bonuses to production speed or efficiency.

**Comment by mirasrael** [comment]
About involving crafters in server activity. As I said before usually you're not only worker or crafter, so while your production is crafting you can go and mine materials or go to build your house or sit into your truck and go to resources. 
But if somebody may be interested only in it's own profession there may be a way to involve player in crafting, in example he can open table UI and work like an manager for his personal (remember invisible elves?) to increase speed at calories cost. It may be supported by mini-game as you suggests. Hmm, may be like drunk man game? :-D Where your table worker goes with cargo to crafting station and you should keep him going with left/right arrows? :)

**Comment by PacifyerGrey** [comment]
> OK. While writing this I have an another idea. What if we will need personal for machines to work? In real world you can't just queue up production for wood working table, you have to get tools and work on it, but making this in Eco as I said before may become very boring and annoying. Instead we can imagine what your table needs personal to work (which is like invisible elves) and you should feed that personal. Then like for some tables you need coal to smelt you will need food. You will not spent calories directly, but instead you will put food into production in other way. To justify high-level food usage it may provide bonuses to production speed or efficiency.

You just mirrored my suggestion here. The minigame part. It requires your presence and attention at the machine. It does not need to involve you in creation of every one of 500 bricks but it should hold you for the length of the whole project.
As an example I gave from FF XIV you should use special crafting actions during a process which will effect the result. Those actions might save you resources or increase project progression and they will consume your calories. These actions might be gated by diet tier etc.

Another example might be that table can craft the item by itself but in order to apply crafting efficiency and crafting speed you must be present and hit those actions on the table

**Comment by mirasrael** [comment]
I understand what you mean, but I don't like an idea when you should wait for 1mx500 = 1000 minutes to produce 500 bricks. I'm personally don't like this kind of gameplay. It may be truly optional when you can participate in craft to increase speed if you don't want to wait too long for craft to finish, but it should work without your constant presence.
Because primary resource for this game is an energy produced from food then it looks logical to consume this energy for almost any production, but converting whole game to table-mini-game doesn't look like an good change for me.

**Comment by mirasrael** [comment]
In real life if you want to scale your production you will hire workers, but because hiring workers for every table almost impossible in Eco (at current stage at least) I suggested to make virtual stuff.

**Comment by PacifyerGrey** [comment]
> What is good and bad with worker professions - almost anyone can do them. And skills for that professions almost useless for now. So everyone has crafting profession and you can't opposite them with working professions.

I have used mods Better Mining and Better Logging as well as I did much of both and can definitely tell those can qualify as main jobs. The problem is that in vanilla skills do not really help you improve but that is another story.

Closer to the endgame so many resources are needed that you just absolutely need to get better in this as well as professional miners and loggers can keep pollution and other stuff under control compared to a random Joe going out for mining.

**Comment by PacifyerGrey** [comment]
> Because primary resource for this game is an energy produced from food then it looks logical to consume this energy for almost any production, but converting whole game to table-mini-game doesn't look like an good change for me.
Take not that minigame is only one of 3 suggestions I made here. Take a look at other 2 if they make sense on their own.

**Comment by mirasrael** [comment]
For me it seems I handled all of them:

> Machines can require a certain food quality of the crafter using them. This is currently reflected by SP/day but you can also split it into tiers and add the number to a diet display. This will drive people to improve their diet to higher tiers driving the whole economy.

While in general it may be an good idea it has some technical problems. It may work for you whole suggestion, but for automated production you should always care if you food level don't go bellow limit and track all tables if they don't stop working. So I suggested alternative: high-tier food may add some benefits to production where it was used (better speed, less calories consumption to production unit etc). 

> Machines can consume calories per minute. This will force crafters use food at all as well as effectively prevent them from super high throughput and almost disable offline activity. This will also make crafting speed skills much more valuable so they can also get a progressive SP costs same as efficiency currently have.

My alternative suggestion was to use food as fuel for crafting. So it doesn't differs much from eating this food and spending these calories by yourself, but allows you to keep this automated (and also closes question - what kind of magic makes these machines work?) And I don't like disabling offline activity. I don't like to be forced to stay in game or keep it open while I'm offline and checking every X minutes to feed my character. It is good thing in Eco what you can build your self-driven business, but it is bad what it doesn't require food like other activities. So adding food fuel will bring it in line with more calorie-expensive professions.

> A general idea to engage a crafter in the crafting process so it would require his own time to progress. Such as some kind of a crafting mini-game at the crafting table. However this might be just too much for a single player but still. This minigame can consume those calories per action making the process look natural (crafter is actually crafting something at the table) as well as make crafting rate limited. A good example might be crafting system in Final Fantasy XIV. Surely that one is too complex for Eco but the idea makes every person be actually engaged in their activity.

My answer was to make this optional. So if you like this you can do it and get benefits, but you shouldn't be forced to do it.

**Comment by PacifyerGrey** [comment]
> My alternative suggestion was to use food as fuel for crafting. So it doesn't differs much from eating this food and spending these calories by yourself, but allows you to keep this automated (and also closes question - what kind of magic makes these machines work?) And I don't like disabling offline activity. I don't like to be forced to stay in game or keep it open while I'm offline and checking every X minutes to feed my character. It is good thing in Eco what you can build your self-driven business, but it is bad what it doesn't require food like other activities. So adding food fuel will bring it in line with more calorie-expensive professions.

I like this option as an alternative. However there are some issues with it still should be accounted in my opinion:

- You have to add visible food tiers to tooltips
- Have to make machines require certain food tier as a fuel
- Actually diet balance is not accounted here which will lead to minmaxing and overproducing one easiest type of food for all the machines removing the variety and causing production chain imbalance. That is why I suggested to use calories from a player. However it would be nice to see top tier food required for endgame machines as top tiers do include almost all production chain materials required.

While being a bit different and less restrictive compared to mine this solution is still an improvement over existing system so I would be glad to see it implemented.
